src/operations/find.ts

```diff
@@ -93,16 +93,6 @@ export class FindOperation extends CommandOperation<Document> {
       throw new MongoInvalidArgumentError('Query filter must be a plain object or ObjectId');
```

Comment:
Should we also be throwing here if we passed a buffer or do we just rely on the server error?

Reply:
I think we should rely on the server error.  What do you think?

If we do throw for binary specifically, we should add guards for each BSON type because our other BSON classes will pass this check: `typeof filter !== 'object' || Array.isArray(filter)`.

Reply:
I would agree with relying on the server error, filters are bson documents -> bson documents to _us_ are plain JS objects / Maps, passing in a Binary is breaking that API contract